hibernate-core/src/main/java/org/hibernate/query/sqm/tuple/internal/AnonymousTupleType.java

```diff
@@ -74,7 +75,10 @@ public AnonymousTupleType(SqmSelectableNode<?>[] components) {
 		componentIndexMap = linkedMapOfSize( components.length );
 		for ( int i = 0; i < components.length; i++ ) {
 			final SqmSelectableNode<?> component = components[i];
-			final String alias = component.getAlias();
+			String alias = aliases == null ? null : aliases.get( i );
+			if ( alias == null ) {
+				alias = component.getAlias();
+			}
```

Comment:
I don't think you need this double-check: you're already getting the selectable node's alias in `extractAliases()`.